packages/datadog-instrumentations/src/cassandra-driver.js

```diff
@@ -0,0 +1,207 @@
+        return wrapCallback(asyncEndCh, errorCh, res)
+      } else {
+        return res.then(
+          () => finish(asyncEndCh, errorCh),
+          err => finish(asyncEndCh, errorCh, err)
+        )
+      }
+    } catch (e) {
+      finish(asyncEndCh, errorCh, e)
+      throw e
+    } finally {
+      endCh.publish(undefined)
+    }
+  })
+
+  shimmer.wrap(cassandra.Client.prototype, 'stream', stream => function (query, params, options, callback) {
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+    const res = stream.apply(this, arguments)
+
+    bindEmit(res, asyncResource)
+    return res
+  })
+  return cassandra
+})
+
+addHook({ name: 'cassandra-driver', versions: ['>=4.4'] }, cassandra => {
+  const startCh = channel('apm:cassandra:_execute:start')
+  const asyncEndCh = channel('apm:cassandra:_execute:async-end')
+  const endCh = channel('apm:cassandra:_execute:end')
+  const errorCh = channel('apm:cassandra:_execute:error')
+  shimmer.wrap(cassandra.Client.prototype, '_execute', _execute => function (query, params, execOptions, callback) {
+    startCh.publish({ client: this, query })
+    const promise = _execute.apply(this, arguments)
+    promise.then(
+      () => finish(asyncEndCh, errorCh),
+      err => finish(asyncEndCh, errorCh, err)
+    )
+    endCh.publish(undefined)
+    return promise
+  })
+  return cassandra
+})
+
+addHook({ name: 'cassandra-driver', versions: ['3 - 4.3'] }, cassandra => {
+  const startCh = channel('apm:cassandra:_innerExecute:start')
+  const asyncEndCh = channel('apm:cassandra:_innerExecute:async-end')
+  const endCh = channel('apm:cassandra:_innerExecute:end')
+  const errorCh = channel('apm:cassandra:_innerExecute:error')
+  shimmer.wrap(cassandra.Client.prototype, '_innerExecute', _innerExecute =>
+    function (query, params, execOptions, callback) {
+      if (!startCh.hasSubscribers) {
+        return _innerExecute.apply(this, arguments)
+      }
+      const asyncResource = new AsyncResource('bound-anonymous-fn')
+      const isValid = (args) => {
+        return args.length === 4 || typeof args[3] === 'function'
+      }
+
+      if (!isValid(arguments)) {
+        return _innerExecute.apply(this, arguments)
+      }
+
+      startCh.publish({ client: this, query })
+
+      const lastIndex = arguments.length - 1
+      let cb = arguments[lastIndex]
+
+      if (typeof cb === 'function') {
+        cb = asyncResource.bind(cb)
+        arguments[lastIndex] = AsyncResource.bind(wrapCallback(asyncEndCh, errorCh, cb))
+      }
+
+      try {
+        return _innerExecute.apply(this, arguments)
+      } catch (e) {
+        finish(asyncEndCh, errorCh, e)
+        throw e
+      } finally {
+        endCh.publish(undefined)
+      }
+    }
+  )
+  return cassandra
+})
+
+addHook({ name: 'cassandra-driver', versions: ['>=3.3'], file: 'lib/request-execution.js' }, RequestExecution => {
+  const addHostCh = channel(`apm:cassandra:general:addHost`)
```

Comment:
`addHost` is what the subscriber does with the message and doesn't accurately represent this operation.

Reply:
It looks like it handles a connection, so maybe it should be `connect` or `connection`, and then the connection options could be passed directly on the message.